src/components/textarea/textarea.test.tsx

```diff
@@ -0,0 +1,719 @@
+  );
+
+  expect(loggerSpy).toHaveBeenCalledTimes(1);
+});
+
+test("should render a textarea element", () => {
+  render(<Textarea />);
+  const textarea = screen.getByRole("textbox");
+  expect(textarea).toBeInTheDocument();
+  expect(textarea.tagName).toEqual("TEXTAREA");
+});
+
+test("should not render character counter if no characterLimit prop is given", () => {
+  render(<Textarea />);
+  expect(screen.queryByText(/you can enter up to/i)).not.toBeInTheDocument();
+});
+
+test("should render character counter if characterLimit prop is given", () => {
+  render(<Textarea characterLimit={5} />);
+  expect(screen.getByText(/you can enter up to/i)).toBeInTheDocument();
+});
+
+test("should not render a placeholder if disabled", () => {
+  render(<Textarea placeholder="foo" disabled />);
+  expect(screen.getByRole("textbox")).toHaveAttribute("placeholder", "");
+});
+
+testStyledSystemMargin((props) => <Textarea {...props} />);
+
+// not sure what this is really testing - I assume it's the expandTextArea stuff, but can't get it to fail!
+// would rather remove this but leave it, commented out, to see if it causes coverage problems. Would like
+// to rewrite if it does, otherwise happy to remove it!
+test("the height of the textarea should remain unchanged when the window is resized", () => {
+  render(<Textarea />);
+
+  const expectedScrollHeight = 500;
+  const originalHeight = "50px";
+  const textarea = screen.getByRole("textbox");
+
+  textarea.style.height = originalHeight;
+  jest
+    .spyOn(textarea, "scrollHeight", "get")
+    .mockImplementation(() => expectedScrollHeight);
+
+  window.dispatchEvent(new Event("resize"));
+
+  expect(textarea.style.height).toEqual(originalHeight);
+});
+
+test("should display the correct value when the value prop is set", () => {
+  render(<Textarea value="Initial content" />);
+
+  expect(screen.getByRole("textbox")).toHaveTextContent("Initial content");
+});
+
+test.each([
+  {
+    inputIcon: "search",
+  },
+  {
+    error: "error",
+  },
+  {
+    warning: "warning",
+  },
+  {
+    info: "info",
+  },
+] as const)(
+  "styles the textarea appropriately when an icon is present inside",
+  (props) => {
+    render(<Textarea value="test string" data-role="icon-test" {...props} />);
+    expect(screen.getByTestId("icon-test")).toHaveStyleRule(
+      "padding-right",
+      "var(--spacing500)",
+      { modifier: `& ${StyledInput}` }
+    );
+  }
+);
+
+test.each([
+  "enter",
+  "done",
+  "go",
+  "next",
+  "previous",
+  "search",
+  "send",
+] as EnterKeyHintTypes[])(
+  "'enterKeyHint' is correctly passed to the input when prop value is '%s'",
+  (keyHint) => {
+    render(<Textarea enterKeyHint={keyHint} />);
+
+    expect(screen.getByRole("textbox")).toHaveAttribute(
+      "enterkeyhint",
+      keyHint
+    );
+  }
+);
+
+const validationTypes = ["error", "warning", "info"];
+
+describe.each(validationTypes)(
```

Comment:
**suggestion**: to minimise the amount of nested tests we have here, what do you think about amending the structure to the following?:

```tsx
it.each(["error", "warning", "info"] as const)("renders validation icon within textarea when %s prop is a string", () => {});
it.each(["error", "warning", "info"] as const)("renders validation icon on the label when %s prop is a string and validationOnLabel is true", () => {});
it("position of validation icon tooltip is overridden, when tooltipPosition prop is set", () => {});
```

This would approach would align with our other test files, ensuring consistency and improving readability.